lib/origin_simulator.ex

```diff
@@ -39,31 +41,52 @@ defmodule OriginSimulator do
     Process.sleep(10)
 
     recipe = Recipe.parse(Plug.Conn.read_body(conn))
-    Simulation.add_recipe(:simulation, recipe)
+    response = Simulation.add_recipe(:simulation, recipe)
+
+    {code, message, content_type} = case response do
+      :ok -> {201, Poison.encode!(Simulation.recipe(:simulation)), "application/json"}
+      :error -> {406, "Not Acceptable", "text/html"}
+    end
 
     conn
-    |> put_resp_content_type("application/json")
-    |> send_resp(201, Poison.encode!(Simulation.recipe(:simulation)))
+    |> put_resp_content_type(content_type)
+    |> send_resp(code, message)
   end
 
   get "/*glob" do
-    serve_payload(conn)
+    recipe_route = Simulation.route(:simulation)
```

Comment:
minor - we could avoid this temporary variable with
```
serve_payload_for_route(conn, serve_payload(conn), conn.request_path)
```
Still quite readable in my opinion

Reply:
👍 shall give the codebase a scan, to see if there are other temporary variables to remove.